pkg/virt-controller/services/nodeselectorrenderer_test.go

```diff
@@ -0,0 +1,145 @@
+package services
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	v1 "kubevirt.io/api/core/v1"
+)
+
+var _ = Describe("Node Selector Renderer", func() {
+	var nsr *NodeSelectorRenderer
+
+	Context("scheduling pods ", func() {
+		When("no selectors are defined in the VMI spec", func() {
+			BeforeEach(func() {
+				nsr = NewNodeSelectorRenderer(emptySelectors(), emptySelectors())
+			})
+
+			It("the node requires the KubeVirt schedulable label", func() {
+				Expect(nsr.Render()).To(Equal(map[string]string{"kubevirt.io/schedulable": "true"}))
+			})
+
+			When("the dedicated CPU option is defined", func() {
+				BeforeEach(func() {
+					nsr = NewNodeSelectorRenderer(emptySelectors(), emptySelectors(), WithDedicatedCPU())
+				})
+
+				It("must be scheduled on nodes featuring the `cpumanager` label", func() {
+					Expect(nsr.Render()).To(HaveKeyWithValue("cpumanager", "true"))
```

Comment:
nit: you can maybe have something like the following function to avoid repeating the "true" part:
```go
hasLabel := func(labelKey string) GomegaMatcher { return HaveKeyWithValue(labelKey, "true") }
```

Reply:
uuuu this one is pure gold. I'll incorporate it asap.

Thanks.